tests/kafkatest/tests/core/replica_scale_test.py

```diff
@@ -0,0 +1,139 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.mark.resource import cluster
+from ducktape.mark import parametrize
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+
+from kafkatest.services.trogdor.produce_bench_workload import ProduceBenchWorkloadService, ProduceBenchWorkloadSpec
+from kafkatest.services.trogdor.consume_bench_workload import ConsumeBenchWorkloadService, ConsumeBenchWorkloadSpec
+from kafkatest.services.trogdor.task_spec import TaskSpec
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.trogdor.trogdor import TrogdorService
+from kafkatest.services.zookeeper import ZookeeperService
+
+import json
+import time
+
+
+class ReplicaScaleTest(Test):
+    def __init__(self, test_context):
+        super(ReplicaScaleTest, self).__init__(test_context=test_context)
+        self.test_context = test_context
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(self.test_context, num_nodes=8, zk=self.zk)
+
+    def setUp(self):
+        self.zk.start()
+        self.kafka.start()
+
+    def teardown(self):
+        # Need to increase the timeout due to partition count
+        for node in self.kafka.nodes:
+            self.kafka.stop_node(node, clean_shutdown=False, timeout_sec=60)
+        self.kafka.stop()
+        self.zk.stop()
+
+    @cluster(num_nodes=12)
+    @parametrize(topic_count=500, partition_count=34, replication_factor=3)
+    def test_produce_consume(self, topic_count, partition_count, replication_factor):
+        t0 = time.time()
+        for i in range(topic_count):
+            topic = "replicas_produce_consume_%d" % i
+            print("Creating topic %s" % topic)  # Force some stdout for Jenkins
+            topic_cfg = {
+                "topic": topic,
+                "partitions": partition_count,
+                "replication-factor": replication_factor,
+                "configs": {"min.insync.replicas": 1}
+            }
+            self.kafka.create_topic(topic_cfg, describe=False)
```

Comment:
Not for this patch, but we should do a KIP to add support for batch topic creation.